src/timetable_gui.cpp

```diff
@@ -87,19 +87,19 @@ static inline TimerGameTick::Ticks TicksPerTimetableUnit()
 
 /**
  * Determine if a vehicle should be shown as late, depending on the timetable display setting.
- * @param v The vehicle in question.
+ * @param ticks How many ticks the vehicle is late.
  * @param round_to_day When using ticks, if we should round up to the nearest day.
  * @return True if the vehicle is later than the threshold.
  */
-bool VehicleIsAboveLatenessThreshold(const Vehicle *v, bool round_to_day)
+bool VehicleIsAboveLatenessThreshold(TimerGameTick::Ticks ticks, bool round_to_day)
 {
 	switch (_settings_client.gui.timetable_mode) {
 		case TimetableMode::Days:
-			return v->lateness_counter > Ticks::DAY_TICKS;
+			return ticks > Ticks::DAY_TICKS;
```

Comment:
Thanks for the contribution!

Why change what is passed to the function, if all that's needed is to take the absolute value of the lateness counter?

If I'm not mistaken, this fix could be confined to `VehicleIsAboveLatenessThreshold()` by simply changing each switch case as below.

```suggestion
			return abs(v->lateness_counter) > Ticks::DAY_TICKS;
```

Reply:
Hi, thank you for your suggestion. The problem here is that is that in this line:
`bool show_late = this->show_expected && VehicleIsAboveLatenessThreshold(v->lateness_counter, true);` it is necessary to actually determine whether the vehicle is late (and not early). With you suggestion show_late would be true for early vehicles as well. I've tried it the way you suggested and that's the issue I ran into. I think it makes sense to make it possible to determine if vehicle is both later/early or just late.

Reply:
Ah, thanks for the explanation. You are correct. 🙂 

Do you mind applying my comment suggestions? Then this is ready for an approval. 😃